### DISCUSSION TITLE
Softphone UA TLS & certificate usage clarification

### DISCUSSION THREAD

**Discussion by renaud-sc** [resiprocate/resiprocate#333] [Q&A]
Hi,

We are trying to add TLS transport to our resip based softphone (Windows and Linux). After fighting with building resip using OpenSSL 3 on Windows (we will soon post some questions/suggestions about that) I am trying to understand the certificates required in our use case. I have read the following documentation page but I am still confused:
- [Using TLS](https://github.com/resiprocate/resiprocate/wiki/Using-TLS)
- [Certificates](https://github.com/resiprocate/resiprocate/wiki/Certificates)
- [TLSAuthentication](https://github.com/resiprocate/resiprocate/wiki/TLSAuthentication)

Our use case is basically a multi-account UA, you could see it as a multiple softphone in one application. In this case, what kind of certificate should we use ? root_cert_, domain_cert_ domain_key, user_cert_ or user_key ? Basically what is the difference or use case of these different certificates ? As we need to support multiple account in one UA, what is the implication on the certificates ?

I believe the documentation would benefit from some clarifcation and use case examples.

Many thanks

**Comment by sgodin** [comment]
If you can do all your SIP traffic to the far end on a single outbound TLS connection, then things are the easiest (then your app is a TLS client only).  However, if you need to support the farend connecting back to you over TLS (or the farend requires MTLS), then you must configure proper server side certificates, ie: your configured certificates will be presented to the far end.  For either case you must have the necessary root certificates on your machine to validate any certificates the far end presents to you.   

None of this is specific to resip, it is standard TLS operation.  resip docs do not try to explain how TLS/MTLS works, which can make them hard to digest if TLS and certificates are a mystery.

Regardless of whether you require MTLS or TLS server side handling or not, resip TLS transports require you to specify the certificates that would be needed.  However if they aren't needed (if you need TLS client side only with no MTLS), then you don't need to worry too much about their content or who they are signed by .  

So what do you need for certs in resip?

Every TLS transport needs:
- domain cert and public key: ie: domain_cert_sip.example.com.pem, domain_key_sip.example.com.pem
    - If you be connected to by others (ie your app is a TLS Server) or far-end requires MTLS, then the SubjectAltName / CommonName must contain the IP address or DNS name that others use to connect to your server
    - if you need client side only (non MTLS), then these certs are loaded, but never need to be presented to the far end, so their contents is less important
- root cert for the signer of your certificates, ie: root_cert_verisign.pem
- root certs to validate any certificates you will receive when forming TLS client side connections to other TLS servers
